src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -2359,6 +2359,52 @@ def get_network_dataplane(self) -> Union[str, None]:
         :return: str or None
         """
         return self.raw_param.get("network_dataplane")
+    
+    def get_acns(self) -> Union[bool, None]:
+        """Get the enablement of acns
+        
+        :return: bool or None"""
+        enable_acns = self.raw_param.get("enable_acns")
+        disable_acns = self.raw_param.get("disable_acns")
+        if enable_acns and disable_acns:
+            raise MutuallyExclusiveArgumentError(
+                "Cannot specify --enable-acns and "
+                "--disable-acns at the same time."
+            )
+        if enable_acns is False and disable_acns is False:
+            return None
+        else:
+            return enable_acns | (not disable_acns)
```

Comment:
this will result in a type error comparing a bool to none. need to separate the checks out

`if enable_acns is not None:
            return enable_acns
        if disable_acns is not None:
            return not disable_acns
        return None`